client/pool.go

```diff
@@ -298,6 +312,9 @@ func (pool *Pool) closeOldIdleConnections() {
 	ticker := time.NewTicker(5 * time.Second)
 
 	for range ticker.C {
+		if pool.ctx.Err() != nil {
```

Comment:
not a big problem. If you change it to

select {
case <- ctx: ...
case <- ticker: ...
}

it will not require 5 seconds to exit at worst.

And if this goroutine is also tracked by pool.wg, we won't have temporary goroutine leak after pool is closed. Some unit test tools may complain about it, like https://github.com/uber-go/goleak

Reply:
@lance6716, PTAL, it's updated according to your suggestion.